src/test/java/edu/bradyreed/advancedjava/ArrayUtilsTest.java

```
@@ -81,4 +84,62 @@ public void testLastIndexOf() {
         
         assertEquals(expectedResult, result);
     }
+    
+    //Begin Negative Tests
+    @Test(expected = BadArrayException.class)
+    public void testExceptionMinValue() throws BadArrayException {
+        int[] testArray = null;
+        int result = -9999;
+        
+        result = ArrayUtils.minValue(testArray);
+             
+}
+    
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testOutOfBoundsCopyRange(){
+        int[] testArray = {5,10,14,1,22,87,16,12,77,46,29,33,-9,17,87,77};
+        int copyPosition = 22;
+        int[] result = null;
+        int[] expectedResult = {87,16,12,77,46,29,33,-9,17,87,77};
+        
+        try {
+            result = ArrayUtils.copyRange(testArray, copyPosition);
+        }
+        catch (BadArrayException e) {}
```

you should not need the try catch here. 
also empty catch blocks (or catch blocks that just log) are a really bad practice as it hides the error. Don't do this.

---

I agree they are a bad practice, i remember learning not to do them. I was surprised to read that this empty catch block was recommended as the solution for building tests that throw exceptions. I assumed since there isnt supposed to be any action taken in the test (other than logging the thrown exception) that it might be ok in this particular case. 

Thoughts?

---

where did you read this?
you can check for exceptions using an annotation like this
 ```java
  @Test(expected = ArithmeticException.class)
    public void testDivisionWithException() {
        int i = 1 / 0;
    }
 ```